addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/OceanicThingHandler.java

```diff
@@ -231,14 +77,49 @@ public void run() {
                     }
                 }
             } catch (Exception e) {
-                logger.error("An exception occurred while polling the Oceanic Water Softener: '{}'", e.getMessage());
+                logger.error("An exception occurred while polling the Oceanic Water Softener: '{}'", e.getMessage(), e);
+                scheduler.schedule(resetRunnable, 0, TimeUnit.SECONDS);
             }
         }
     };
 
+    private Runnable resetRunnable = new Runnable() {
```

Comment:
Looks the same as reconnectRunnable, should it be executed with a delay to make sure we are not overloading g the machine once we are in an error state?

Reply:
I don't think so, the resetRunnable is only schedule()d once, but I have altered the code to include a  updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());

Reply:
Ok, thanks